packages/react/src/SearchControl/SearchControl.tsx

```diff
@@ -577,15 +577,19 @@ function getPage(search: SearchRequest): number {
 
 function getTotalPages(search: SearchRequest, lastResult: Bundle): number {
   const pageSize = search.count ?? DEFAULT_SEARCH_COUNT;
-  return Math.ceil(getTotal(search, lastResult) / pageSize);
+  const total = getTotal(search, lastResult);
+  return total > 0 ? Math.ceil(total / pageSize) : 1;
 }
 
 function getStart(search: SearchRequest, lastResult: Bundle): number {
   return Math.min(getTotal(search, lastResult), (search.offset ?? 0) + 1);
 }
 
 function getEnd(search: SearchRequest, lastResult: Bundle): number {
-  return Math.min(getTotal(search, lastResult), ((search.offset ?? 0) + 1) * (search.count ?? DEFAULT_SEARCH_COUNT));
+  return Math.min(
```

Comment:
What was the change here?

I don't have any memory of why this function needs to call `getTotal()`... Is it necessary here?

Reply:
The main change: 
```ts
- ((search.offset ?? 0) + 1) * (search.count ?? DEFAULT_SEARCH_COUNT)
+ getStart(search, lastResult) + (search.count ?? DEFAULT_SEARCH_COUNT) - 1
``` 

The multiplication logic was incorrect for larger numbers: 

e.g. 
- pageSize: 20
- offset 100K
- total: 200K

Before: 
end = (100K) * (20) = 2M

After:
end = 100K + 20 = 100,020

Reply:
getTotal() is needed in this context: 

count: 20
offset 10
total: 25


Desired output: 
start=10
end=25 (not 30)